cpp/include/cudf/table/row_operators.cuh

```diff
@@ -78,10 +78,6 @@ __device__ weak_ordering compare_elements(Element lhs, Element rhs)
  * `[-Inf, -ve, 0, -0, +ve, +Inf, NaN, NaN, null] (for null_order::AFTER)`
  * `[null, -Inf, -ve, 0, -0, +ve, +Inf, NaN, NaN] (for null_order::BEFORE)`
  *
- * @param lhs first element
- * @param rhs second element
- * @return Indicates the relationship between the elements in
- * the `lhs` and `rhs` columns.
```

Comment:
doxygen throws warning that params are documented twice because of SFINAE. These arguments are documented in other definition.
`warning: argument 'lhs' from the argument list of cudf::relational_compare has multiple @param documentation sections`

Both `@brief` will show in the documentation. https://docs.rapids.ai/api/libcudf/nightly/namespacecudf.html#a9b480332cf7484e8532a2d6f2c711a35

Reply:
So, honestly, any functions that requires SFINAE should probably not be in a public header. That is, I cannot picture a non-internal libcudf API that would have SFINAE semantics. And there has already been discussion on moving the row-operators into the `detail` namespace. In general, I would not spend too much time on trying to get doxygen working with SFINAE'd functions.

Reply:
That is a fair point. I guess the question then would be, should doxygen be linting those at all? It's a bit tricky right now because we also use `detail` to hold the stream versions of public APIs, and those may eventually become public, but in the longer run we probably want to tell doxygen either not to lint those files or to at least relax some requirements. 

Anyway, I'm fine with this PR not resolving that question.